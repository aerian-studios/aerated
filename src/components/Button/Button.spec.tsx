import React from "react";

import { render, screen } from "@testing-library/react";

import { Button } from "./index";

describe("Button", () => {
    it("should render correctly", () => {
        const { container } = render(<Button disabled={false} />);

        expect(screen.getByRole(/button/)).toBeInTheDocument();
        expect(screen.getByRole(/button/)).not.toBeDisabled();
        expect(container).toMatchSnapshot();
    });

    it("should be disabled when passed the disabled prop", () => {
        const { container } = render(<Button disabled={true} />);

        expect(screen.getByRole(/button/)).toBeInTheDocument();
        expect(screen.getByRole(/button/)).toBeDisabled();
        expect(container).toMatchSnapshot();
    });

    it("should pass button correct variant classnames", () => {
        const variants = ["dark", "tertiary", "negative"] as const;

        variants.forEach((variant: typeof variants[number]) => {
            const { container } = render(<Button variant={variant} />);
            expect(screen.getByRole(/button/)).toHaveClass(variant);
            container.remove();
        });
    });
});
