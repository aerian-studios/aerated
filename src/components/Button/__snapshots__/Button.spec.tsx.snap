// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be disabled when passed the disabled prop 1`] = `
<div>
  <button
    class="component"
    disabled=""
  />
</div>
`;

exports[`Button should render correctly 1`] = `
<div>
  <button
    class="component"
  />
</div>
`;
